virttest/qemu_vm.py

```diff
@@ -994,16 +994,10 @@ def set_value(opt_string, key, fallback=None):
                 set_value("port=%s", "spice_port")
 
             set_value("password=%s", "spice_password", "disable-ticketing")
-            if optget("listening_addr") == "ipv4":
-                host_ip = utils_net.get_host_ip_address(self.params)
-                self.spice_options['listening_addr'] = "ipv4"
+            if optget("listening_addr"):
+                host_ip = utils_net.get_host_ip_address(
+                    self.params, ip_version=optget("listening_addr"))
```

Comment:
Make it more obvious:

ip_ver = optget("listening_addr")
if ip_ver:
    ost_ip = utils_net.get_host_ip_address(self.params, ip_version=ip_ver)